test/model.test.js

```diff
@@ -0,0 +1,893 @@
+'use strict';
+
+var Promise = require('bluebird'),
+    TestFixture = require('./test-fixture'),
+    expect = require('chai').expect,
+    errors = require('../lib/errors'),
+    util = require('./util');
+
+var test = new TestFixture(),
+    thinky, type;
+
+before(function() {
+  return test.initializeDatabase()
+    .then(function() {
+      thinky = test.thinky;
+      type = thinky.type;
+    });
+});
+after(function() { return test.dropDatabase(); });
+
+describe('createModel', function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should create a new model', function() {
+    var model = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    return model.ready()
+      .then(function() { expect(model).to.exist; });
+  });
+
+  it('should check if the table was created', function() {
+    var modelName = util.s8();
+    var model = thinky.createModel(modelName, {
+      id: String, name: String
+    });
+
+    return model.ready()
+      .then(function() {
+        return test.r.tableList().run();
+      })
+      .then(function(result) {
+        expect(result).to.contain(modelName);
+      });
+  });
+
+  it('should create multiple models', function() {
+    var model1 = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    var model2 = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    return Promise.all([ model1.ready(), model2.ready() ])
+      .then(function() {
+        expect(model1).to.not.eql(model2);
+      });
+  });
+
+  it('should throw error for non existent table access', function() {
+    var model = thinky.createModel("nonExistentTable", {
+      id: String, name: String
+    }, {
+      init: false
+    });
+
+    return expect(model.get(1).run())
+      .to.eventually.be.rejectedWith(/^Table `.*` does not exist in/);
+  });
+});
+
+describe('[_]getModel', function() {
+  it('_getModel', function() {
+    var name = util.s8();
+    var model = thinky.createModel(util.s8(), {
+      id: String, name: String
+    }, {
+      init: false
+    });
+
+    expect(model._getModel()).to.have.property('_name');
```

Comment:
As far as I can tell from a small test, this is equivalent to

```
assert('_name' in model._getModel()))
```

Which is not the same as

```
assert(model._getModel().hasOwnProperty('name')))
```